src/twisted/web/server.py

```diff
@@ -500,6 +505,27 @@ def getRootURL(self):
         return self.appRootURL
 
 
+    def _handleStar(self):
+        """
+        Handle receiving a request whose path is '*'.
+
+        RFC 7231 defines an OPTIONS * request as being something that a client
+        can send as a low-effort way to probe server capabilities or readiness.
+        Rather than bother the user with this, we simply fast-path it back to
+        an empty 200 OK. Any non-OPTIONS verb gets a 405 Method Not Allowed
+        telling the client they can only use OPTIONS.
+        """
+        if self.method == b'OPTIONS':
+            self.setResponseCode(http.OK)
+        else:
+            self.setResponseCode(http.NOT_ALLOWED)
+            self.setHeader(b'Allow', b'OPTIONS')
+
```

Comment:
I think we're squarely in the realm of interpretation here.

My reading of 7231 is that we could go either to 400 or 405. My interpretation is that, given that the request is sensibly parseable and understandable, 400 is not really quite right. But I don't believe this strongly: I almost wrote the patch as a 400.

Regardless, I don't really think the user should get a look. It's never something to which you can sensibly send a response to.

So I guess what I'm saying is that I can be talked around to 400 pretty easily.